### ISSUE TITLE
YOLO11 Freezing layer 'model.23.dfl.conv.weight'

### ISSUE THREAD

**Issue by hinakurumi** [ultralytics/ultralytics#17202]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

When I loaded the yolo11l-seg.pt pre-trained model to train my own dataset, I adjusted self.reg_max to 32 to solve the issue of elongated object bounding boxes not being fully covered. However, I found that YOLO11 by default freezes model.23.dfl.conv.weight. After unfreezing model.23.dfl.conv.weight, the following warning message appeared during training:

"WARNING ⚠️ setting 'requires_grad=True' for frozen layer 'model.23.dfl.conv.weight'. See ultralytics.engine.trainer for customization of frozen layers."

So I would like to know if this modification has any negative impact.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @hinakurumi, thank you for reaching out and for using Ultralytics 🚀! This is an automated response, but no worries—an Ultralytics engineer will assist you soon.

We recommend visiting our [Docs](https://docs.ultralytics.com) where new users can find a plethora of [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples that may address your needs.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us diagnose and resolve the issue effectively.

If you have a custom training ❓ Question, similar to your query about modifying `model.23.dfl.conv.weight`, please supply as much context as possible. Include dataset examples and training logs to assist us, and ensure you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best! For real-time chat, head over to [Discord](https://ultralytics.com/discord) 🎧, engage in deeper discussions on [Discourse](https://community.ultralytics.com), or explore community threads on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Please upgrade to the latest `ultralytics` package, ensuring all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) are met in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO runs smoothly in the following verified environments, complete with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

This badge indicates our continuous integration (CI) status. Green means all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests for macOS, Windows, and Ubuntu are passing, verifying proper function of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/).

Thank you for your patience and support! We look forward to helping you resolve your issue.

**Comment by Y-T-G** [comment]
It's supposed to be kept frozen.

**Comment by Sanster** [comment]
I checked the original official implementation, and I think the `class DFL` in ultralytics corresponds to the original `Project` layer (which is not trainable) https://github.com/implus/GFocal/blob/master/mmdet/ops/project.py#L5.

**Comment by hinakurumi** [comment]
Thanks!



发自我的iPhone


------------------ Original ------------------
From: Glenn Jocher ***@***.***&gt;
Date: Fri,Nov 8,2024 7:36 PM
To: ultralytics/ultralytics ***@***.***&gt;
Cc: hinakurumi ***@***.***&gt;, Mention ***@***.***&gt;
Subject: Re: [ultralytics/ultralytics] YOLO11 Freezing layer'model.23.dfl.conv.weight' (Issue #17202)





 
Thank you for pointing that out. You're correct; the DFL class in Ultralytics corresponds to the non-trainable Project layer in the original implementation.

**Comment by glenn-jocher** [comment]
You're welcome! If you have any more questions or need further assistance with YOLO11, feel free to ask.
